packages/eslint-plugin-react-hooks/src/RulesOfHooks.js

```diff
@@ -526,6 +526,15 @@ export default {
             reactHooksMap.set(codePathSegment, reactHooks);
           }
           reactHooks.push(node.callee);
+        } 
+        if (node.arguments.some(isHook)) {
+          node.arguments.filter(isHook).forEach(hook => {
+            const message =
+              `React Hook "${context.getSource(hook)}" cannot be called ` +
```

Comment:
This is a bit misleading. We can't claim it's being called inside a callback — it's being _passed_ to a callback. Can we amend this?

Reply:
Does this sound better?

```suggestion
              `React Hook "${context.getSource(hook)}" cannot be passed ` +
              'as a callback. React Hooks must be called in a ' +
              'React function component or a custom React Hook function.';
```